### ISSUE TITLE
Read binary files (wav files) from sqlite

### ISSUE THREAD

**Issue by MircoBerlin** [1bpm/csound-sqldb#1]
@1bpm I tested the opcode(s) in cabbage. And was succesful reading some strings from it and print them. 

I also tested to fetch some data (wav files) that are stored in the db, but without luck yet.
Is it already supported?

Thank you!

**Comment by 1bpm** [comment]
Reading binary data isn't supported - just numeric/string data or data that can be converted to those. What you have mentioned sounds interesting, I had a quick look at how it could be done, and it could involve a fair bit of work so not something I could have functional for a while. 
How would you imagine it being used, eg getting a direct audio output from a database query, or loading to a GEN1 ftable perhaps?

This has reminded me to push to this repository - I host at [http://git.1bpm.net/csound-sqldb/](http://git.1bpm.net/csound-sqldb/) and forgot about github - there has been a bug fix since the last push here, involving a memory leak with queries returning arrays.

**Comment by MircoBerlin** [comment]
Yeah pushing it to a table or to the diskin opcodes would be great. Maybe I should try to find a way to convert my wav files to strings.

Also sqlite can handle password protection which would be great feature.

**Comment by 1bpm** [comment]
I think it would be hard to make work at the moment - at a push it may be possible to cast the blob to hex and somehow read sample values out with `dbarray` then write to a table, but would be tricky getting sample format etc. Can't think of anything that could use the blob cast to string returned from the database, so it would really need some new opcodes writing.

Can't find anything about a standard for password protection/encryption for SQLite [apart from SEE](https://sqlite.org/com/see.html) which isn't open source, how are you encrypting your database?

**Comment by MircoBerlin** [comment]
Here are some interesting suggestions: https://stackoverflow.com/questions/1381264/password-protect-a-sqlite-db-is-it-possible

**Comment by 1bpm** [comment]
I read that very page actually. The only viable option mentioned there 
is SQLCipher - the others are either .NET specific, or not open source.

On 2022-10-15 12:24, MircoBerlin wrote:

> Here are some interesting suggestions: 
> https://stackoverflow.com/questions/1381264/password-protect-a-sqlite-db-is-it-possible
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://github.com/1bpm/csound-sqldb/issues/1#issuecomment-1279724821
[2]
